### DISCUSSION TITLE
r.learn.ml x r.learn.ml2

### DISCUSSION THREAD

**Discussion by CarlosGrohmann** [OSGeo/grass#1855] [Q&A] [answered]
Sorry if this is too obvious, but what are the differences between r.learn.ml and r.learn.ml2? From the codes, it looks like r.learn.ml is the older version and r.learn.ml2 is newer, but that is not clear from the manual pages. If r.learn.ml is deprecated, why not keep only the newer r.learn.ml2? 
thanks

**Comment by veroandreo** [comment, marked as answer]
> Sorry if this is too obvious, but what are the differences between r.learn.ml and r.learn.ml2? From the codes, it looks like r.learn.ml is the older version and r.learn.ml2 is newer, but that is not clear from the manual pages. If r.learn.ml is deprecated, why not keep only the newer r.learn.ml2?
> thanks

r.learn.ml* are addons and rules are not so strict there, afaiu. See related discussion in the original PR in the addons repo: https://github.com/OSGeo/grass-addons/pull/126 for pointers on why the 2 versions coexist

After one year, maybe it's worth considering dropping the old version and making r.learn.ml2 --> the new r.learn.ml (@stevenpawley?). In any case, you can just use the one you like best ;-) 

I'd suggest to open an issue in the addons repo

**Reply by CarlosGrohmann** [reply]
Thanks, Vero! Now I see the reason for keeping both versions

**Reply by stevenpawley** [reply]
Thanks Vero for providing the context for this.

I agree that the version 1 should probably be dropped now. v2 has been in use for a while now, but relating to the confusion that Carlos highlighted, I think quite a few people are using/downloading version 1.

What is the convention within GRASS for deprecating modules?

**Reply by veroandreo** [reply]
I have not found anything updated about deprecating addons; only [this](https://trac.osgeo.org/grass/wiki/HowToSVN#MovingmodulesbetweentrunkandAddonsrepository), but it is oldish and moreover about moving addons to core. 

Maybe, once the upcoming grass8 branch is created in addons repo, you could remove version 1 there. So, version 1 and 2 would be available in grass7 addons branch and only version 2 in grass8 branch. Just a suggestion, dunno however if that's a good idea. Let's see what others say :) 

In any case, a deprecation warning should be included somehow in version 1 of the module.
